NString.Tests/StringTemplateConverterTests.cs

```diff
@@ -0,0 +1,309 @@
+﻿using System.Globalization;
+using System.Linq;
+using Xunit;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace NString.Tests
+{
+    public class StringTemplateConverterTests
+    {
+        private static readonly string[] StringCollectionValues = { "1", "2", "3" };
+        private const string ExpectedStringCollectionConcatenation = "1|2|3";
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Property()
+        {
+            var values = new BasicTestValues { StringCollectionProperty = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{StringCollectionProperty}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Field()
+        {
+            var values = new BasicTestValues { StringCollectionField = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{StringCollectionField}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Value_Type()
+        {
+            var values = new BasicTestValues { ValueTypeProperty = new DateTime(ticks: 42) };
+
+            string actual = StringTemplate.Format("{ValueTypeProperty}", values);
+
+            Assert.Equal("42", actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Works_With_FormatProvider()
+        {
+            var values = new ValuesForFormatProvider { DateTimeConvertedWithFortyTwoDaysAdded = DateTime.Today };
+            var expectedConvertedDate = values.DateTimeConvertedWithFortyTwoDaysAdded + TimeSpan.FromDays(42);
+
+            var cultures = new[] { CultureInfo.InvariantCulture, null, new CultureInfo("fr-FR") };
+
+            foreach (var culture in cultures)
+            {
+                string expected = expectedConvertedDate.ToString("D", culture);
+                string actual = StringTemplate.Format("{DateTimeConvertedWithFortyTwoDaysAdded:D}", values, true, culture);
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Works_With_FormatProvider_And_Return_Type_Change()
+        {
+            var ticks = 1337;
+            var values = new ValuesForFormatProvider { DateTimeConvertedToTicks = new DateTime(ticks) };
+            var formatProvider = new NumberFormatInfo { CurrencySymbol = "piasses", CurrencyPositivePattern = 3 /* symbol at the end, with a space */ };
+            var expectedHexTicks = string.Format(formatProvider, "{0:C0}", ticks);
```

Comment:
Hex ? You should probably rename the variable :wink:

Reply:
Oops! Sorry about that! Do you mind if I just amend my last commit?

Reply:
Sure!